trunner/target/armv7a9.py

```diff
@@ -56,7 +56,7 @@ def __call__(self):
             try:
                 subprocess.run(
                     ["gdb-multiarch", "plo-gdb.elf", "-x", self.script, "-batch"],
-                    encoding="ascii",
+                    encoding="charmap",
```

Comment:
why change of encoding? Seems like trying to hide some other errors...  (as charmap-based encoding will accept any byte, will just fail at a later time - when trying to decode into UTF-8).

Also - I'm not sure it's valid without explicitly stating some charmap (like `cp1252`) - do we have non-standard locale somewhere resulting in national chars being printed?

Reply:
> why change of encoding? Seems like trying to hide some other errors... (as charmap-based encoding will accept any byte, will just fail at a later time - when trying to decode into UTF-8).
> 
> Also - I'm not sure it's valid without explicitly stating some charmap (like `cp1252`) - do we have non-standard locale somewhere resulting in national chars being printed?

Yes, that's not perfect, but I'd opt for leaving `charmap`, or rather `cp1252` to provide the full output - that unfortunately contains non-utf8 char. From what I see it's only used in case of an errror and then it's appended to an exception (this specific char is transformed, but that's not problematic in my opinion):
![Screenshot from 2023-10-18 11-40-00](https://github.com/phoenix-rtos/phoenix-rtos-tests/assets/77304431/0c2ed540-81d9-4645-ad62-9cba55396849)

What is the exact case, where it may cause some problems?

Removing `.decode("ascii")` from `raise FlashError(msg=str(e), output=e.stdout.decode("ascii") if e.stdout else None) from e` would also be needed.

Reply:
Soooo, this hides some underlying, not diagnosed problem without any comment.

IMHO we should try to pin-point (and hopefully fix) underlying issue instead of hiding it like that.

- does it happen also in non-batch (interactive) mode?
- which command results in erroneous char being printed?
- is there a global .gdbinit which may change it?
- is it related to reset not being configured?
- does gdb version change the output (eg. using `arm-none-eabi-gdb` from GCC toolchain?)

Reply:
Yeah, for sure the comment was missing. Thanks for suggestions, you are right, it's worth to check. @maska989 could you follow them?

Reply:
Problem solved via setting up reset_config and tweeks into upload-zynq7000-smt2.gdb.

Reply:
great to hear that :)